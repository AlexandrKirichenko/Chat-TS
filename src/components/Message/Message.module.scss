@import "../../constants";

.wrapMessage {
  display: flex;
  //align-items: center;
  &.reverse {
    flex-direction: row-reverse;
  }
  
  .messageBlock {
    max-width: 745px;
    background-color: #01C5C4;
    box-shadow: $box-shadow;
    border-radius: $border-radius-me;
    background-color:$color-input-border;
    margin:7px 0;
    &.notMe {
      border-radius:16px 4px 16px 4px;
      background-color:$color-secondary;
    
    }
  }
}


