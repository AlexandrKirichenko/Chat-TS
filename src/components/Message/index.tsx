import classnames from 'classnames'
import React from 'react'
import Avatar from '../Avatar'
import style from './Message.module.scss';
import './Message.module.scss';

export interface MessageProp {
    id: string,
    messageText: string,
    userId: number,
    login: string;
    avatar?: string;
    itsMe: boolean;
}

const Message: React.FC<MessageProp> = ({itsMe, avatar, login, messageText}) => {
    
    return (
        <div data-testid="message__block" className={classnames(style.wrapMessage, {[style.secondary]: itsMe})}>
            <Avatar value={avatar} nameAvatar={login} sizeAvatar={'small'}/>
            <div data-testid="message" className={classnames(style.messageBlock, {[style.secondary]: itsMe})}>
                {messageText}
            </div>
        </div>
    )
}

export default Message;

