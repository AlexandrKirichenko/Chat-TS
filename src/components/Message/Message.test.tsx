import React from "react";
import Message, {MessageProp} from "./index";
import {create, act} from "react-test-renderer";
import TestRenderer from 'react-test-renderer';

const messageText =
    "Lorem1 ipsum dolor sit amet, consectetur adipisicing elit. Accusamus aperiam beatae consequatur " +
    "cupiditate delectus dolore est id maxime mollitia necessitatibus nemo, neque optio quidem quis recusandae sint sunt " +
    "tempore vitae?";

const defaultProps1: MessageProp = {
    itsMe: false,
    login: "login 1",
    avatar: "",
    messageText: messageText,
    userId: 1,
    id: "1",
};

const defaultProps2: MessageProp = {
    itsMe: false,
    login: "login 2",
    avatar: "",
    messageText: messageText,
    userId: 2,
    id: "2",
};

let container: any;

describe("Message component", () => {
    it("renders correctly", () => {
        act(() => {
            container = create(<Message {...defaultProps1} itsMe={true}/>).toJSON();
            expect(container).toMatchSnapshot();
        });
    });
    
    it.each([
        [defaultProps1],
        [defaultProps2]
    
    ])('dinamyc styles not adds when flag true', (defaultProps)=>{
        const testRenderer = TestRenderer.create(<Message {...defaultProps} itsMe={false}/>);
        const testInstance = testRenderer.root;
        expect(testInstance.findByProps({"data-testid": "message__block"}).props['className']).toBe('wrapMessage');
        expect(testInstance.findByProps({"data-testid": "message"}).props['className']).toBe('messageBlock');
    })
    
    it.each([
        [defaultProps1],
        [defaultProps2]
    
    ])('dinamyc styles not adds when flag false', (defaultProps)=>{
        const testRenderer = TestRenderer.create(<Message {...defaultProps} itsMe={false}/>);
        const testInstance = testRenderer.root;
        expect(testInstance.findByProps({"data-testid": "message__block"}).props['className']).toBe('wrapMessage');
        expect(testInstance.findByProps({"data-testid": "message"}).props['className']).toBe('messageBlock');
    })
});








