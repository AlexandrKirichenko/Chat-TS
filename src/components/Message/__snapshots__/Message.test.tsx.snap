// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component Message rendering 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapMessage"
        data-testid="message__block"
      >
        <div
          class="wrap"
        >
          <img
            class="avatar small"
            src="https://avatarfiles.alphacoders.com/798/79894.jpg"
          />
        </div>
        <div
          class="messageBlock"
          data-testid="message"
        >
          Lorem1 ipsum dolor sit amet, consectetur adipisicing elit. Accusamus aperiam beatae consequatur cupiditate delectus dolore est id maxime mollitia necessitatibus nemo, neque optio quidem quis recusandae sint sunt tempore vitae?
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapMessage"
      data-testid="message__block"
    >
      <div
        class="wrap"
      >
        <img
          class="avatar small"
          src="https://avatarfiles.alphacoders.com/798/79894.jpg"
        />
      </div>
      <div
        class="messageBlock"
        data-testid="message"
      >
        Lorem1 ipsum dolor sit amet, consectetur adipisicing elit. Accusamus aperiam beatae consequatur cupiditate delectus dolore est id maxime mollitia necessitatibus nemo, neque optio quidem quis recusandae sint sunt tempore vitae?
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Message component renders correctly 1`] = `null`;
